content/chapters/io/lab/content/file-mappings.md

```diff
@@ -54,6 +54,26 @@ mmap(0x7fb313e84000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENY
 For a quick reminder about `mmap()`, its 5th argument is the file descriptor from where we want to copy the data to the RAM.
 The 6th argument is the offset within the file from where to start copying.
 
+### What About `read()`
+
+Let's get the elephant out of the room: why use `mmap()` instead of `read()` for dynamic libraries?
+
+The short answer: for efficiency and convenience.
+
+Getting a dynamic library into memory using `read()` might seem similar to `mmap()`-ing it, but it does not take advantage of how these libraries are used.
+Dynamic libraries are **non-writable** areas in the process VAS, that are **arbitrarily accessed**.
```

Comment:
This is too coarse. They also contain writable sections (like `.data`) that are subject to copy-on-write.

Reply:
My bad.